src/app/shared/document/document-detail/document-detail.component.ts

```diff
@@ -17,379 +17,128 @@
-      .forEach(row => {
-        dirty = true;
-        document.data[row.id] = row.value;
-      });
-
-    return dirty;
-  }
-
-  // patch those that are defined in collection but were unused
-  private patchReusedAttributes(document: DocumentModel): boolean {
-    let dirty = false;
-    this.rows.filter(row =>
-      isNullOrUndefined(row.id) && row.name &&
-      this.alreadyInCollection(row.name))
-      .forEach(row => {
-        dirty = true;
-        let attr = this.getCollectionAttributeByName(row.name);
-
-        row.id = attr.id;
-        row.correlationId = undefined;
-        document.data[attr.id] = row.value;
-      });
-
-    return dirty;
-  }
-
-  private patchAttributeRename(document: DocumentModel): boolean {
-    let dirty = false;
-
-    this.rows.filter(row =>
-      !isNullOrUndefined(row.id) &&
-      row.name !== this.getCollectionAttributeById(row.id).name)
-      .forEach(row => {
-        dirty = true;
-        const attr = this.getCollectionAttributeByName(row.name);
-
-        if (attr) { // renamed to existing attribute
-          if (!document.data[attr.id]) {
-            delete document.data[row.id];
-            document.data[attr.id] = row.value;
-            row.id = attr.id;
-            row.correlationId = undefined;
-          } else if (document.data[attr.id] && this.getRemovableRowByName(row.name) && this.getRowByName(row.name)) {
-            const originalRow = this.getRemovableRowByName(row.name);
-            originalRow.id = row.id;
-            document.data[attr.id] = row.value;
-            row.id = attr.id;
-            row.correlationId = undefined;
-          }
-        } else { // renamed to a completely new name
-          delete document.data[row.id];
-          row.id = null;
-          row.correlationId = CorrelationIdGenerator.generate();
-          document.newData[row.name] = { value: row.value, correlationId: row.correlationId };
-        }
-      });
-
-    return dirty;
-  }
-
-  private patchDeletedAttributes(document: DocumentModel): boolean {
-    let dirty = false;
-
-    this.collection.attributes.forEach(attr => {
-      if (document.data[attr.id] && this.getRowById(attr.id) && this.getRowById(attr.id).remove) {
-        dirty = true;
-        delete document.data[attr.id];
-      }
-    });
-
-    return dirty;
-  }
-
-  private getCollectionAttributeById(id: string): AttributeModel {
-    return this.collection.attributes.find(attr => attr.id === id);
+    this.renewSubscriptions();
   }
 
-  private getCollectionAttributeByName(name: string): AttributeModel {
-    return this.collection.attributes.find(attr => attr.name === name);
+  public ngOnChanges(changes: SimpleChanges): void {
+    this.renewSubscriptions();
   }
 
-  private patchDocument() {
-    const updatedDocument = this.prepareUpdatedDocument();
+  private renewSubscriptions(): void {
+    this.subscriptions.unsubscribe();
+    this.documentUiService.destroy(this.collection, this.document);
```

Comment:
Are you sure to reinit `DocumentUi` after each change? What happens to the new data that was generated during the server response?

Reply:
That is a good question, I need to reinvestigate, because the goal was to use it only during change of document or collection.